builtin/logical/totp/path_keys.go

```diff
@@ -0,0 +1,399 @@
+package totp
+
+import (
+	"bytes"
+	"encoding/base32"
+	"encoding/base64"
+	"fmt"
+	"image/png"
+	"net/url"
+	"strconv"
+	"strings"
+
+	"github.com/hashicorp/vault/logical"
+	"github.com/hashicorp/vault/logical/framework"
+	otplib "github.com/pquerna/otp"
+	totplib "github.com/pquerna/otp/totp"
+)
+
+func pathListKeys(b *backend) *framework.Path {
+	return &framework.Path{
+		Pattern: "keys/?$",
+
+		Callbacks: map[logical.Operation]framework.OperationFunc{
+			logical.ListOperation: b.pathKeyList,
+		},
+
+		HelpSynopsis:    pathKeyHelpSyn,
+		HelpDescription: pathKeyHelpDesc,
+	}
+}
+
+func pathKeys(b *backend) *framework.Path {
+	return &framework.Path{
+		Pattern: "keys/" + framework.GenericNameRegex("name"),
+		Fields: map[string]*framework.FieldSchema{
+			"name": {
+				Type:        framework.TypeString,
+				Description: "Name of the key.",
+			},
+
+			"generate": {
+				Type:        framework.TypeBool,
+				Default:     false,
```

Comment:
I guess it depends on how most Vault users would use the backend. If most people use it as a "client" and store an external key it would make sense to have the default be false. If you think that more people would be using it as a "server" and generating keys to be sent out we could change it to true.